src/main/java/io/supertokens/oauth/OAuth.java

```diff
@@ -461,50 +469,51 @@ public static void verifyAndUpdateIntrospectRefreshTokenPayload(Main main, AppId
             payload.entrySet().clear();
             payload.addProperty("active", false);
 
-            // // ideally we want to revoke the refresh token in hydra, but we can't since we don't have the client secret here
-            // refreshToken = refreshToken.replace("st_rt_", "ory_rt_");
-            // Map<String, String> formFields = new HashMap<>();
-            // formFields.put("token", refreshToken);
+            refreshToken = refreshToken.replace("st_rt_", "ory_rt_");
+            Map<String, String> formFields = new HashMap<>();
+            formFields.put("token", refreshToken);
 
-            // try {
-            //     doOAuthProxyFormPOST(
-            //         main, appIdentifier, oauthStorage,
-            //         clientId, // clientIdToCheck
-            //         "/oauth2/revoke", // path
-            //         false, // proxyToAdmin
-            //         false, // camelToSnakeCaseConversion
-            //         formFields,
-            //         new HashMap<>());
-            // } catch (OAuthAPIException | OAuthClientNotFoundException e) {
-            //     // ignore
-            // }
+            try {
+                OAuthClient oAuthClient = OAuth.getOAuthClientById(main, appIdentifier, storage, clientId);
+                formFields.put("client_secret", oAuthClient.clientSecret);
+                formFields.put("client_id", oAuthClient.clientId);
+
+                HttpRequestForOAuthProvider.Response revokeResponse = doOAuthProxyFormPOST(
+                     main, appIdentifier, oauthStorage,
+                     clientId, // clientIdToCheck
+                     "/oauth2/revoke", // path
+                     false, // proxyToAdmin
+                     false, // camelToSnakeCaseConversion
+                     formFields,
+                     new HashMap<>());
+
+            } catch (OAuthAPIException | OAuthClientNotFoundException | InvalidKeyException | NoSuchAlgorithmException |
+                    InvalidKeySpecException | NoSuchPaddingException | InvalidAlgorithmParameterException |
+                    IllegalBlockSizeException | BadPaddingException e){
+                //ignore
+            }
         }
     }
 
     private static boolean isTokenRevokedBasedOnPayload(OAuthStorage oauthStorage, AppIdentifier appIdentifier, JsonObject payload) throws StorageQueryException {
```

Comment:
I think this should become just checking if it is revoked based on gid+jti, the rest of the checks no longer make sense, since they fail at exactly the same time now.

Reply:
except for the we still need a method for checking by the row in the db. I opted for the ..GID version to keep.